File: Data Structures & Algorithms/top-k-elements-in-list/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let n = nums.length;
        const arr = Array.from({ length: n + 1 }, () => []);
        const count = {};

        for (const num of nums) {
            count[num] = (count[num] || 0) + 1;
        }

        for (const num in count) {
            arr[count[num]].push(parseInt(num));
        }


        const res = [];
        for (let i = arr.length - 1; i > 0; i--) {
            for (const n of arr[i]) {
                res.push(n);

                if (res.length === k) {
                    return res;
                }
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/sort-colors/submission-6.java
class Solution {
    public void sortColors(int[] nums) {
        int l = 0, r = nums.length - 1;
        int i = 0;

        while(i <= r){
            if(nums[i] == 0){
                swap(nums ,i, l);
                l++;
            }else if(nums[i] == 2){
                swap(nums,i ,r);
                r--;
                i--;
            }

            i++;
        }
    }

    private void swap(int[] nums, int i, int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

// [2,0,2,1,1,0]   -> [0,0,2,1,1,2] 
// [il, , , , ,r]  -> [l,i, , ,r, ]

// [0,0,2,1,1,2]  -> [0,0,2,1,1,2]
// [l,i, , ,r, ]  -> [,l,i , ,r, ]

// [0,0,2,1,1,2] -> []

